tests/apps/search/test_templatetags_search_tags.py

```diff
@@ -0,0 +1,41 @@
+"""
+Unit tests for the template tags of the search app.
+"""
+from django.core.exceptions import ImproperlyConfigured
+from django.test import TestCase
+from django.test.utils import override_settings
+
+from richie.apps.search.templatetags.search_tags import react_locale
+
+
+@override_settings(REACT_LOCALES=["en_US", "fr_CA", "fr_FR", "es_ES"])
+class ReactLocaleSearchTemplateTagsTestCase(TestCase):
+    """
+    Unit test suite to validate the behavior of the `react_locale` template filter.
+    """
+
+    def test_templatetags_search_tags_react_locale_full_language(self):
+        """
+        Full languages should return the equivalent React locale.
+        """
+        self.assertEqual(react_locale("fr-FR"), "fr_FR")
+        self.assertEqual(react_locale("fr-CA"), "fr_CA")
+
+    def test_templatetags_search_tags_react_locale_simple_language(self):
+        """
+        Simple languages should return the first matching React locale.
+        """
+        self.assertEqual(react_locale("en"), "en_US")
+        self.assertEqual(react_locale("es"), "es_ES")
+        self.assertEqual(react_locale("fr"), "fr_CA")
```

Comment:
There is a mistake the code `fr` can not be the canadian version of french. It's the "true" french.

Reply:
Ok I understadn your test. You should maybe add an other one with the same settings eith `fr_FR` before `fr_CA` ?

Reply:
ok